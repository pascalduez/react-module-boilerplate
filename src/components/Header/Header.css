.root {
  color: orangeRed;
  background-color: var(--color-gray);
}

.title {
  margin: 0;
}
