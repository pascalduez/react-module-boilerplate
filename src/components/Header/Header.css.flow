// @flow

declare module.exports: $ReadOnly<{
  root: string,
  title: string,
}>;
