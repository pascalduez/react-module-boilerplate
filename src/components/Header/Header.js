// @flow

import * as React from 'react';
import clsx from 'clsx';

import styles from './Header.css';

export type Props = {
  title: string,
  className?: string,
  ...
};

function Header({ title, className, ...extraProps }: Props) {
  return (
    <header
      role="banner"
      className={clsx(styles.root, className)}
      {...extraProps}
    >
      <h1 className={styles.title}>{title}</h1>
    </header>
  );
}

export default Header;
