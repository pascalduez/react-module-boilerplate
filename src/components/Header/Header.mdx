---
name: Header
---

import { Playground, PropsTable } from 'docz';
import Header from './Header';

# Header

## Basic usage

<Playground>
  <Header title="Hello" />
  <Header title="ðŸ˜€ ðŸ˜Ž" />
</Playground>

## PropTypes

<PropsTable of={Header} />
