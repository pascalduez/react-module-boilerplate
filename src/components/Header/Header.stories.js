// @flow

import * as React from 'react';

import Header from './Header';

export default {
  component: Header,
  title: 'Header',
};

export const text = () => <Header title="Hello" />;

export const emoji = () => <Header title="ðŸ˜€ ðŸ˜Ž" />;
