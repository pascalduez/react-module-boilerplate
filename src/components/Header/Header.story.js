// @flow

import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { checkA11y } from '@storybook/addon-a11y';

import Header from './Header';

storiesOf('Header', module)
  .addDecorator(withKnobs)
  .addDecorator(checkA11y)
  .add('with text', () => <Header title={text('Title', 'Hello')} />)
  .add('with some emoji', () => <Header title="ðŸ˜€ ðŸ˜Ž" />);
