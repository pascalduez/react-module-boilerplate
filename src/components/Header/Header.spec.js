// @flow

import { render } from '@testing-library/react';

import Header from './Header';
import styles from './Header.css';

describe('<Header>', () => {
  it('should render the default markup', () => {
    const { container, getByText } = render(<Header title="Test" />);

    const title = getByText('Test');

    expect((container.firstChild: any).tagName).toBe('HEADER');
    expect(title.tagName).toBe('H1');
  });

  it('should render the default classes', () => {
    const { container, getByText } = render(<Header title="Test" />);

    const title = getByText('Test');

    expect(container.firstChild).toHaveClass(styles.root);
    expect(title).toHaveClass(styles.title);
  });

  // Which is equivalent to:

  it('should render the default markup and classes [snapshot]', () => {
    const { container } = render(<Header title="Test" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
