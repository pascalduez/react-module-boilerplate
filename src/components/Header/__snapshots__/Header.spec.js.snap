// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render the default markup and classes [snapshot] 1`] = `
<header
  class="Header-root"
  role="banner"
>
  <h1
    class="Header-title"
  >
    Test
  </h1>
</header>
`;
